### ISSUE TITLE
Add an option like `display_with` but only which takes a `&self`

### ISSUE THREAD

**Issue by zhiburt** [zhiburt/tabled#13]
It may be useful as sometimes we would like to print a content on the basis of some internal information not only on the type information itself.

```rust
#[derive(Tabled)]
struct House {
   #[field(display_with_self = "foo")]
   key: String,
}

fn foo(&House) -> String { ... }
```

**Comment by zhiburt** [comment]
The name isn't perfect though :)

**Comment by zhiburt** [comment]
Could be done by an parameter in attribute

```rust
    #[tabled(display_with(with_self, "display_based_on"))]
```

**Comment by zhiburt** [comment]
Closed by #169
